src/main/java/core/basesyntax/db/Storage.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.db;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private static final Map<String, Integer> fruitStorage = new HashMap<>();
+
+    public static void setBalance(String fruit, int quantity) {
+        if (quantity < 0) {
+            throw new RuntimeException("Quantity can't be negative");
+        }
+        if (fruitStorage.values().stream().anyMatch(value -> value < 0)) {
+            throw new RuntimeException("Balance can't be negative");
+        }
+        fruitStorage.put(fruit, quantity);
+    }
```

Comment:
This method does not follow the encapsulation principle. The fruitStorage map should not be modified directly outside of the Storage class. Instead, create instance methods for operations like set, add, and remove, and make the fruitStorage map private.